### PR TITLE
Improve how the project root is resolved

### PR DESCRIPTION
**Pull request by ehmicky**
As described in https://github.com/netlify/cli/issues/859, this improves how the project root is resolved, by moving the `findRoot` logic to `@netlify/config` instead.

Note: this PR is a WIP since it requires https://github.com/netlify/build/pull/1579 and https://github.com/netlify/build/pull/1580 to be merged first. This is why the tests are currently failing.

### CODE REVIEW

File: src/utils/command.js [hunk 1]
```diff
@@ -23,20 +22,19 @@ class BaseCommand extends Command {
...
     // Grab netlify API token
     const authViaFlag = getAuthArg(argv)
 
     const [token] = this.getConfigToken(authViaFlag)
 
+    const cachedConfig = await this.getConfig(cwd, state, token)
+    const { configPath, config, repositoryRoot: projectRoot } = cachedConfig
+
     // Get site id & build state
     const state = new StateConfig(projectRoot)
```
Comment (erezrokah) [inline src/utils/command.js:36]: Let's say I have the following directory structure:
```
repo/
  .git/
  site/
    .netlify/state.json
    netlify.toml
```

`repositoryRoot` will evaluate to `repo`, am I correct?
That makes sense.

But `StateConfig` tries to read `<projectRoot>/.netlify/state.json` which will fail.

The CLI seems to link a folder to a site (and not a project), thus I'll need to link both `repo` and `repo/site` folders for that to work.

Looks like we should separate concerns between evaluating the repository root and finding the location of `state.json`

Comment (ehmicky) [reply src/utils/command.js:36]: This should now work with https://github.com/netlify/build/pull/1581.

Re-writing the PR accordingly right now.

**Comment by ehmicky** [comment]
I have updated this PR based on https://github.com/netlify/cli/issues/859#issuecomment-649639490.

**Comment by ehmicky** [comment]
There seem to be still be some issues, debugging right now.

**Comment by ehmicky** [comment]
Problems fixed!

**Comment by erezrokah** [comment]
Tested and the fix works.
Also - the code looks much simpler now, thanks @ehmicky!

**Comment by JamesMcMahon** [comment]
Does this change result in breaking existing configuration?

My configuration, which was:
```
[build]
  publish = "build"
```

Broke post netlify cli, as it no longer is checking for `build` relative to where I am running the deploy command. It now expects `build` to be off of the repo root.

Not sure if this was the change that broke it, but looking at the changes it looked like the most likely culprit.

**Comment by ehmicky** [comment]
Hi @JamesMcMahon,

Could you please print your logs when running the Netlify CLI, with the environment variable `NETLIFY_BUILD_DEBUG` set to `true`? Thanks!

**Comment by FinnWoelm** [comment]
@JamesMcMahon, I ran into the same issue. I solved it by creating an empty `.git` directory in my subfolder. For some reason, that worked.

My files:

```
+ .git/
+ node_modules/
+ package.json
+ file1.js
+ file2.js
+ subfolder/
  + .git/    <-- completely empty directory
  + package.json
  + netlify.toml
  + publish/
  + ...
```

My `netlify.toml` configuration:

```
[build]
  publish = "publish"
```

My subfolder package.json:
```
{
  "name": "test",
  "scripts": {
    "deploy": "../../../node_modules/.bin/netlify deploy"
  }
}
```

Perhaps it helps someone :blush:

**Comment by JamesMcMahon** [comment]
Just to clarify, I had no problems fixing the issue. I was reporting it hear because it was a breaking change that got bundled into a seem-lying innocuous minor release.

In my case, I fixed it by changing

```toml
[build]
publish = "build"
```

To:

```toml
[build]
publish = "frontend/build"
```

This is maybe the wrong place to report this, also maybe kind of pointless given that release has already happened. I wanted to record the change for posterity sake.

@ehmicky If you still need any of that information let me know.

**Comment by ehmicky** [comment]
@JamesMcMahon If you could post any of the information listed above, this could fix this problem for other users without them having to modify their `netlify.toml` nor add a `.git` directory. Otherwise, I'm glad your problem is solved :)

**Pull request merged**